pkgs/development/tools/revive/default.nix

```diff
@@ -0,0 +1,49 @@
+{ buildGoModule, fetchFromGitHub, lib }:
+
+buildGoModule rec {
+  pname = "revive";
+  version = "1.2.1";
+
+  src = fetchFromGitHub {
+    owner = "mgechev";
+    repo = "revive";
+    rev = "v${version}";
+    sha256 = "sha256-V0YWp4zQduW+cnuiVoEAPxDnsA4OR0TnJdoOF9TdRjw=";
+  };
+
+  ldflags = [
+    "-X"
+    "github.com/mgechev/revive/cli.version=${version}"
+    "-X"
+    "github.com/mgechev/revive/cli.date=2022-06-22"
+    "-X"
+    "github.com/mgechev/revive/cli.commit=23828cc39d4f91b69247bdea0ab4903bd6c1912e"
+    "-X"
+    "github.com/mgechev/revive/cli.builtBy=nixpkgs"
+  ];
```

Comment:
```suggestion
  ldflags = [
    "-s"
    "-w"
    "-X github.com/mgechev/revive/cli.version=v${version}"
    "-X github.com/mgechev/revive/cli.builtBy=nix"
  ];
```

`-s` and `-w` for smaller build

put `-X` and value in line

I'd strongly recommend against putting commit and date by hand as anyone else updating will miss it

You can either leave commit and date out
Leave commit but set date to 1970-01-01
Or add the date and commit like so:

In `fetchFromGitHub`:
```
    # populate values that require us to use git. By doing this in postFetch we
    # can delete .git afterwards and maintain better reproducibility of the src.
    leaveDotGit = true;
    postFetch = ''
      cd "$out"
      git rev-parse HEAD > $out/COMMIT
      # 0000-00-00T00:00:00Z
      date -u -d "@$(git log -1 --pretty=%ct)" "'+%Y-%m-%d %H:%M UTC'" > $out/SOURCE_DATE_EPOCH
      find "$out" -name .git -print0 | xargs -0 rm -rf
    '';
```
and

```
  # ldflags based on metadata from git and source
  preBuild = ''
    ldflags+=" -X github.com/anchore/grype/internal/version.gitCommit=$(cat COMMIT)"
    ldflags+=" -X github.com/anchore/grype/internal/version.buildDate=$(cat SOURCE_DATE_EPOCH)"
  '';
```



Also please note I changed the date format since what you added was different to the makefile

https://github.com/mgechev/revive/blob/master/Makefile#L7

Also the git tag has a v prefix

https://github.com/mgechev/revive/blob/master/Makefile#L6

And since their value for `builtBy` is `makefile` maybe it should be `nix` rather than `nixpkgs` since the build tool is `nix`

Reply:
I very much appreciate the detailed response, thank you ❤️ ! I will fix this :)

Reply:
Fixed, thank you again for walking me through this!